### ISSUE TITLE
mGBA PSVita Core

### ISSUE THREAD

**Issue by ToniBC** [libretro/mgba#153]
I have not seen the PSVita core of this emulator for a long time and it seems strange to me.

In PSP if it is, in Vita we have it officially, but in Retroarch in the Vita version it does not come out.

I do not know if it is due to a bot compilation error that has already happened sometimes or to other errors, but it is a kernel that seems to work the same or better than VBA Next and it would be good to include it.

Thank you.

**Comment by endrift** [comment]
There is one already. Is it still broken? If so it's been broken for years at this point...

**Comment by ToniBC** [comment]
Installed version 2019-06-27_RetroArch.vpk and still does not appear. It may be an error in the compilation, but that's why I put it, it seemed strange to me.

**Comment by olokos** [comment]
I can confirm this. 
As of 2019-07-21 there was still no mGBA core for retroarch and currently mGBA is the best performing core for GBA on vita. This includes both performance and graphics improvements over VBANext or GBSP. Basically makes me go mGBA for vita natively instead of retroarch for GBA games.

**Comment by ToniBC** [comment]
It would be nice to have the core available, so as not to have to use the standalone version, because in other platforms it is compiled and in Vita for some reason no.

**Comment by hizzlekizzle** [comment]
@frangarcj may be able to explain what's going on with it. I didn't see any activity related to it on the buildbot logs...

**Comment by olokos** [comment]
I think @twinaphex could also tell us what is the reason behind this, especially since he's the leader here.

**Comment by twinaphex** [comment]
I'll go through some of the logs.

**Comment by twinaphex** [comment]
I see the recipe has been disabled. It likely broke down at some point and because when one single core fails to build for statically linked platforms, the entire RetroArch build fails (for all cores), it was likely high priority to get it to work again, so this recipe was forcibly disabled. We'd have to manually check if it finally builds again, and if it doesn't, we'd perhaps have to ask @endrift for guidance.

**Comment by endrift** [comment]
I seem to recall it built but crashed. Don't know why. I think that's what bug #69 was about.

**Comment by docterd** [comment]
Any chance to revisit this topic?

**Comment by frangarcj** [comment]
I think it is fixed.

**Comment by endrift** [comment]
Can you confirm that? I know there was a memory leak in platforms/psp2/memory.c, which got copy/pasted into platforms/libretro/memory.c when @twinaphex insisted that libretro only use one memory.c file, so unless that mistake ever got corrected, then that leak will still be there too.

For the record, I still don't know what the rationale of using that file instead of just using the existing ones was.

**Comment by frangarcj** [comment]
My fix was a workaround then. I will use malloc and free for Vita too.

**Comment by docterd** [comment]
I'm on a retroarch nightly build based on a2a5814 (end october) and it is not included.

**Comment by endrift** [comment]
What @frangarcj meant was that the bug that was why the decision to disable the core was fixed, but the core didn't get re-enabled.

Anyway I talked to him on Discord and we agreed that the Vita-specific code in `libretro/memory.c` should just be removed outright, due to how newlib is being used. Using calloc (not malloc; that was an error on my part which I just fixed upstream) and free should be a decent default for all OSes, but having the ability for a given Makefile to override it with the existing platform specific version (e.g. having the win32 version of the core use the win32 memory.c) would probably be a better way to handle that.
